src/prefect/flow_engine.py

```diff
@@ -633,11 +688,12 @@ def cancel_all_tasks(self):
     @contextmanager
     def start(self) -> Generator[None, None, None]:
         with self.initialize_run():
-            self.begin_run()
+            with trace.use_span(self._span):
+                self.begin_run()
 
-            if self.state.is_running():
-                self.call_hooks()
-            yield
+                if self.state.is_running():
+                    self.call_hooks()
+                yield
```

Comment:
stylistic nit:
```python
with self.initialize_run(), trace.use_span(self._span):
    self.begin_run()
    if self.state.is_running():
        self.call_hooks()
    yield
```

Reply:
dang i was tempted to do this everytime i looked at the diff 😄